plaso/parsers/sqlite_plugins/safari.py

```diff
@@ -0,0 +1,160 @@
+  DESCRIPTION = 'Parser for Safari history SQLite database files.'
+
+  #Define the needed queries.
+
+  QUERIES = [
+      (('SELECT history_items.id, history_items.url, history_items.visit'
+        '_count, history_visits.id AS visit_id, history_visits.history_item,'
+        'history_visits.visit_time, history_visits.redirect_destination, '
+        'history_visits.title, history_visits.http_non_get, '
+        'history_visits.redirect_source, history_items.domain_expansion '
+        'FROM history_items, history_visits '
+        'WHERE history_items.id = history_visits.history_item '
+        'ORDER BY history_visits.visit_time'), 'ParsePageVisitRow')
+  ]
+
+  # The required tables
+  REQUIRED_TABLES = frozenset(['history_items', 'history_visits'])
+
+  SCHEMAS = [{
+      'history_items': (
+          'CREATE TABLE history_items (id INTEGER PRIMARY KEY AUTOINCREMENT, '
+          'url TEXT NOT NULL UNIQUE,domain_expansion TEXT NULL, '
+          'visit_count INTEGER NOT NULL, daily_visit_counts BLOB NOT NULL, '
+          'weekly_visit_counts BLOB NULL, autocomplete_triggers BLOB NULL, '
+          'should_recompute_derived_visit_counts INTEGER NOT NULL, '
+          'visit_count_score INTEGER NOT NULL)'),
+      'history_tombstones': (
+          'CREATE TABLE history_tombstones (id INTEGER PRIMARY KEY '
+          'AUTOINCREMENT, start_time REAL NOT NULL, end_time REAL NOT NULL, '
+          'url TEXT,generation INTEGER NOT NULL DEFAULT 0)'),
+      'metadata': ('CREATE TABLE metadata (key TEXT NOT NULL UNIQUE, value)'),
+      'history_client_versions': (
+          'CREATE TABLE history_client_versions (client_version '
+          'INTEGER PRIMARY KEY, last_seen REAL NOT NULL)'),
+      'history_event_listeners': (
+          'CREATE TABLE history_event_listeners (listener_name '
+          'TEXT PRIMARY KEY, last_seen REAL NOT NULL)'),
+      'history_events': (
+          'CREATE TABLE history_events (id INTEGER PRIMARY KEY AUTOINCREMENT,'
+          ' event_type TEXT NOT NULL, event_time REAL NOT NULL, '
+          'pending_listeners TEXT NOT NULL, value BLOB)'
+      ),
+      'history_visits': (
+          'CREATE TABLE history_visits (id INTEGER PRIMARY KEY AUTOINCREMENT,'
+          ' history_item INTEGER NOT NULL REFERENCES history_items(id) ON '
+          'DELETE CASCADE, visit_time REAL NOT NULL, title TEXT '
+          'NULL,load_successful BOOLEAN NOT NULL DEFAULT 1, http_non_get '
+          'BOOLEAN NOT NULL DEFAULT 0, synthesized BOOLEAN NOT NULL DEFAULT '
+          '0, redirect_source INTEGER NULL UNIQUE REFERENCES '
+          'history_visits(id) ON DELETE CASCADE, redirect_destination INTEGER'
+          ' NULL UNIQUE REFERENCES history_visits(id) ON DELETE CASCADE, '
+          'origin INTEGER NOT NULL DEFAULT 0, generation INTEGER NOT NULL '
+          'DEFAULT 0, attributes INTEGER NOT NULL DEFAULT 0, score INTEGER '
+          'NOT NULL DEFAULT 0)'
+      )
+  }]
+
+  def _GetHostname(self, url):
```

Comment:
i had it there because all the other browser history parser plugins have a hostname event data. however , there is a domain expansion field in the database i can use as the hostname so instead of the hostname coming out as www.facebook.com, it will come out as just facebook.